src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -263,3 +274,56 @@ object NonNullableSetter {
     }
   }
 }
+
+object NonNullableIterableSetter {
+  implicit def encodeNonNullableSetterArr[T](
+      implicit encodeT: Encoder[T]
+  ): Encoder[NonNullableIterableSetter[Seq[T]]] = new Encoder[NonNullableIterableSetter[Seq[T]]] {
+    final def apply(a: NonNullableIterableSetter[Seq[T]]): Json = a match {
+      case SetValue(value) =>
+        Json.obj(("set", Json.arr(value.map(encodeT.apply): _*)))
+      case AddRemoveArr(add: Seq[T], remove: Seq[T]) =>
+        Json.obj(
+          ("add" -> Json.arr(add.map(encodeT.apply): _*)),
```

Comment:
Neat! I think we can remove the surrounding parenthesis here? `"foo" -> 12.3` is typically written without them.